src/main/java/IP/duke/main/Storage.java

```diff
@@ -0,0 +1,116 @@
+package IP.duke.main;
+
+import IP.duke.task.Deadline;
+import IP.duke.task.Event;
+import IP.duke.task.Task;
+import IP.duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents a file to store data, operations to write on the file.
+ * 
+ * @author Gordon
+ * @version CS2103T, Semester 2
+ */
+public class Storage {
+    private String filePath;
+    private String fileName;
+    private File file;
+    private FileWriter fileWriter;
+    
+    /**
+     * Class constructor.
+     * 
+     * @param filePath the path from the project directory to the storage file.
+     * @param fileName the name of the file.
+     */
+    public Storage(String filePath, String fileName) {
+        this.filePath = filePath;
+        this.fileName = fileName;
+    }
+
+    /**
+     * Loads all the tasks saved in the storage folder into a tasks arraylist.
+     * 
+     * @return an arraylist of tasks.
+     * @throws IOException
+     */
+    public  ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            File fileDirectory = new File(filePath);
+            if(!fileDirectory.exists()) {
+                fileDirectory.mkdir();
+            }
+            file = new File(filePath + "/" + fileName);
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            if (file.length() != 0) {
+                Scanner sc = new Scanner(file);
+                String line;
+                while (sc.hasNext()) {
+                    line = sc.nextLine();
+                    int START = 8;
+                    Task t;
+                    String divider = " | ";
+                    int taskAndTimeDivider = line.indexOf(divider);
+                    int startingIndex = line.indexOf(divider) + divider.length() + 4;
+                    if (line.contains("D |")) {
+                        String taskDescriptionAndTime = line.substring(startingIndex);
+                        int startOfTimeIndex = taskDescriptionAndTime.indexOf(divider);
+                        String task = taskDescriptionAndTime.substring(0, startOfTimeIndex);
+                        String time = taskDescriptionAndTime.substring(startOfTimeIndex + divider.length());
+                        t = new Deadline(task, time);
+                    } else if (line.contains("E |")) {
+                        String taskDescriptionAndTime = line.substring(startingIndex);
+                        int startOfTimeIndex = taskDescriptionAndTime.indexOf(divider);
```

Comment:
I'm not very sure which is more efficient, but it seems like the code would look shorter if you can extract the 4 common lines for "D |" and "E |" into another if statement block.

also, I think `String taskDescriptionAndTime = line.substring(startingIndex);` could be extracted out of this 3 blocks since it is common to all conditionals! 👯